### PR TITLE
code reorganization and cleanup

### PR DESCRIPTION
**Pull request by zemberdotnet**
# Description

This PR:
- updates dependencies
- removes dead code
- reorganizes the active-directory steps
- various related changes

### CODE REVIEW

File: src/steps/active-directory/__tests__/converters.test.ts [hunk 1]
```diff
@@ -98,26 +99,6 @@ describe('createUserEntity', () => {
   });
 });
 
-describe('createOrganizationEntity', () => {
```
Comment (adam-in-ict) [inline src/steps/active-directory/__tests__/converters.test.ts:101]: Is this test not needed anymore?  I might've missed where this was moved to or what changes make it no longer needed.

Comment (zemberdotnet) [reply src/steps/active-directory/__tests__/converters.test.ts:101]: Ah I forgot to call this out. We don't create Organization entities anywhere in the project, so this was just dead code.

File: src/ms-graph/__tests__/__recordings__/fetchMetadata_2104579518/recording.har [hunk 2]
```diff
@@ -1,317 +0,0 @@
-{
-  "log": {
```
Comment (zemberdotnet) [inline src/ms-graph/__tests__/__recordings__/fetchMetadata_2104579518/recording.har:2]: Removed as `fetchMetadata` is dead code.

File: src/ms-graph/client.ts [hunk 3]
```diff
@@ -67,10 +67,6 @@ export class GraphClient {
     }
   }
 
-  public async fetchMetadata(): Promise<object> {
```
Comment (zemberdotnet) [inline src/ms-graph/client.ts:70]: Dead code.

File: src/steps/active-directory/__tests__/client.test.ts [hunk 4]
```diff
@@ -32,7 +32,8 @@ describe('iterateGroupMembers', () => {
     client = new DirectoryGraphClient(logger, config);
   });
 
-  test('iterateGroupMembers', async () => {
+  // TODO @zemberdotnet Rerecord test
+  test.skip('iterateGroupMembers', async () => {
```
Comment (zemberdotnet) [inline src/steps/active-directory/__tests__/client.test.ts:36]: Test disabled due to the change to page size. Added to my ticket to rerecord tests for this project.

**Review by adam-in-ict — APPROVED** [review]

**Pull request merged**